### ISSUE TITLE
Inventory deadlock

### ISSUE THREAD

**Issue by MedicOP** [ronancpl/HeavenMS#310]
Same as #279, seems the fix was sadly not successful

```java
10-01-2019 11:30:47
    sun.misc.Unsafe.park(Native Method)
    java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
    java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
    java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
    java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
    java.util.concurrent.locks.ReentrantLock$FairSync.lock(ReentrantLock.java:224)
    java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
    net.server.audit.locks.active.TrackerReentrantLock.lock(TrackerReentrantLock.java:77)
    client.inventory.MapleInventory.list(MapleInventory.java:93)
    client.MapleCharacter$10.run(MapleCharacter.java:2770)
    server.TimerManager$LoggingSaveRunnable.run(TimerManager.java:148)
    java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
    java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
    java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    java.lang.Thread.run(Thread.java:748)
```

**Comment by ronancpl** [comment]
Actually, what can be said about that attempt is that'd _potentially solve one_ deadlock potential point.

```
chrRLock.lock();
        objectWLock.lock();
        try {
            int curOID = getUsableOID();
            mapobject.setObjectId(curOID);
            this.mapobjects.put(curOID, mapobject);
            for (MapleCharacter chr : characters) {
                if (condition == null || condition.canSpawn(chr)) {
                    if (chr.getPosition().distanceSq(mapobject.getPosition()) <= getRangedDistance()) {
                        inRangeCharacters.add(chr);
                        chr.addVisibleMapObject(mapobject);
                    }
                }
            }
        } finally {
            objectWLock.unlock();
            chrRLock.unlock();
        }
        
        for (MapleCharacter chr : inRangeCharacters) {
            packetbakery.sendPackets(chr.getClient());
        }
```

Scoping _spawnAndAddRangedMapObject_: objectWLock, chrRLock nows get to lock only to get a list of players in range, instead of run extra code from _sendPackets_.

Thus, indeed, there are more code sections having potential deadlocks to look out, regarding one or more of the picked locks by that thread.

**Comment by ronancpl** [comment]
Issue merged with #316.
